Add pagination to page mentions and page phrases pages

sophiebrown01 opened the pull request:
This PR adds pagination to the pages visited page, the page phrases page, as well as updating the table styles for the sortable table to remove zebra striping and make it visually consistent with a reguar `.govuk-table`.

This PR also includes some other changes, such as reducing the number of pages and mentions on the phrase detail page.

Commit:
Reduce number of pages and mentions on phrase detail page

This commit reduces the number of pages and mentions on the phrase detail page. By doing this we don't overload the user with info, keeping the page as a summary page and also keeping to the design.

Commit:
Add pagination and update table styles

This commit adds pagination to the pages visited page, as well as updating the table styles for the sortable table to remove zebra striping and make it visually consistent with a reguar `.govuk-table`.

Commit:
Link to GOV.UK pages on the phrase detail page

Commit:
Add link to GOV.UK pages on pages visited page

Commit:
Add pagination to phrase usage page

Commit:
Update shown date ranges and remove last updated dates

This commit updates the shown date ranges for the data currently loaded into the app. It also removes last updated dates.

oscarwyatt commented on app/helpers/phrase_usage_helper.rb:
```diff
@@ -2,4 +2,30 @@ module PhraseUsageHelper
+  def phrase_usage_navigation_links
+    links = {}
+
+    if @presenter.pagination.page > 1
```
The pattern I've almost always with helpers is to pass objects in, so to fit this style it should be `def phrase_usage_navigation_links(presenter)` and then use the presenter without the `@` in the method

sophiebrown01 replied:
I can refactor this one out for all pages.

oscarwyatt commented on app/helpers/phrase_usage_helper.rb:
```diff
@@ -2,4 +2,30 @@ module PhraseUsageHelper
+      )
+    end
+
+    if @presenter.pagination.page < @presenter.pagination.total_pages
```
You're violating the law of demeter here, the helper knows too much about the presenter (and the presenter hardly does anything (assuming I'm right in thinking that you're expecting an instance of PhraseUsagePresenter). I'd refactor so the presenter has methods like `page`,`total_pages` etc and the helper only calls `page` or `total_pages` rather than `presenter.pagination.page`

sophiebrown01 replied:
This is the same pattern as we currently use elsewhere in the app (i.e. the SummaryPresenter). The presenter _does_ have these methods as you describe, but they're on the PaginationPresenter which is included in the SummaryPresenter.

I'm also following the same pattern here as Content Data: 

https://github.com/alphagov/content-data-admin/blob/master/app/presenters/content_items_presenter.rb

https://github.com/alphagov/content-data-admin/blob/master/app/presenters/pagination_presenter.rb 

The only difference is their ContentItemsPresenter `delegates` to the other presenters.

oscarwyatt replied:
> The presenter does have these methods as you describe, but they're on the PaginationPresenter which is included in the SummaryPresenter.

The helper should neither know nor care how the summary presenter "knows" which page they're on, so calling `.pagination.foo` means that the helper knows this. I'd suggest you SummaryPresenter should delegate these things directly to the PaginationPresenter, then the helper knows less (plus the code is slightly shorter 🎉 )

oscarwyatt reviewed (changes requested):
Looks good but a couple of minor rails convention-y type things that could be tidied up a bit  😄

Commit:
Refactor navigation_links into application helper

This commit refactors `navigation_links` into application helper and uses it throughout the app where pagination is required.

oscarwyatt reviewed (approved):
Lovin yo work 💯
